From 32735e32f0777155cad03dbd0d45a1d62e14a1af Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 9 Jan 2025 08:38:15 +0100
Subject: [PATCH 1432/1695] plat-stm32mp2: disable tamper and pwr regulator
 drivers in M33TD

Disable flags CFG_STM32_PWR_REGUL and CFG_STM32_TAMP in M33TD
since PWR regulators and tampers are handle only by TDCID.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 6e693c61663587a4cc9dc64e31f6fcfbcc81db2f plat-stm32mp2: disable tamper and pwr regulator drivers in M33TD]
Change-Id: Iac641bc09c222a65c46c9aa0c8d7d697686813cd
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/427744
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 4ae2b7897..ba962b058 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -97,6 +97,8 @@ $(call force,CFG_SCMI_SERVER_PD_CONSUMER,n)
 $(call force,CFG_SCMI_SERVER_REGULATOR_CONSUMER,n)
 $(call force,CFG_SCMI_SERVER_RESET_CONSUMER,n)
 $(call force,CFG_STM32_BSEC_WRITE,n)
+$(call force,CFG_STM32_PWR_REGUL,n)
+$(call force,CFG_STM32_TAMP,n)
 $(call force,CFG_STM32MP_PROVISIONING,n)
 $(call force,CFG_TA_STM32MP_NVMEM,n)
 $(call force,CFG_STM32MP2_CLK_CAL,n)
@@ -121,7 +123,7 @@ $(call force,CFG_SECURE_TIME_SOURCE_CNTPCT,y)
 $(call force,CFG_STM32_SHARED_IO,y)
 $(call force,CFG_STM32_HSE_MONITORING,y)
 $(call force,CFG_STM32_PWR,y)
-$(call force,CFG_STM32_PWR_REGUL,y)
+CFG_STM32_PWR_REGUL ?= y
 $(call force,CFG_STM32MP_CLK_CORE,y)
 CFG_STM32MP2_CLK_CAL ?= y
 CFG_STM32MP_REMOTEPROC ?= y
-- 
2.47.2

